src/test/java/com/google/sps/lib/algorithm/distance/MapsApiDistanceCalculatorTest.java

```diff
@@ -37,10 +40,46 @@
+
+  @Test(expected = RuntimeException.class)
+  public void findDistance_failureResponse_noApiResponse_RuntimeException() throws Exception {
+    when(mockUrlWrapper.openConnection(any())).thenReturn(mockConnection);
+    when(mockConnection.getResponseCode()).thenReturn(200);
+    String apiResponse = "";
+    when(mockConnection.getInputStream()).thenReturn(new ByteArrayInputStream(apiResponse.getBytes("UTF-8")));
+    List<Coordinate> coordinates = new ArrayList<>(Arrays.asList(new Coordinate(28.6129, 77.2295),
+        new Coordinate(22.5726, 88.3639), new Coordinate(19.0760, 72.8777)));
+
+    new MapsApiDistanceCalculator(mockUrlWrapper).findDistance(coordinates);
+  }
+
+  @Test
+  public void findDistance_successresponse_requestUrlAsExpected() throws Exception {
+    when(mockUrlWrapper.openConnection(any())).thenReturn(mockConnection);
+    when(mockConnection.getResponseCode()).thenReturn(200);
+    String apiResponse = "{ \"destination_addresses\" : [ \"Amar Jawan Jyoti Rd, India Gate, New Delhi, Delhi 110001, India\", \"24, Ramakanta Mistri Ln, Cossipore, Newland, College Square, Kolkata, West Bengal 700073, India\", \"1777, Lower Parel, Friends Colony, Kurla West, Kurla, Mumbai, Maharashtra 400070, India\" ], \"origin_addresses\" : [ \"Amar Jawan Jyoti Rd, India Gate, New Delhi, Delhi 110001, India\", \"24, Ramakanta Mistri Ln, Cossipore, Newland, College Square, Kolkata, West Bengal 700073, India\", \"1777, Lower Parel, Friends Colony, Kurla West, Kurla, Mumbai, Maharashtra 400070, India\" ], \"rows\" : [ { \"elements\" : [ { \"distance\" : { \"text\" : \"1 ft\", \"value\" : 0 }, \"duration\" : { \"text\" : \"1 min\", \"value\" : 0 }, \"status\" : \"OK\" }, { \"distance\" : { \"text\" : \"939 mi\", \"value\" : 1510850 }, \"duration\" : { \"text\" : \"1 day 2 hours\", \"value\" : 93607 }, \"status\" : \"OK\" }, { \"distance\" : { \"text\" : \"873 mi\", \"value\" : 1404782 }, \"duration\" : { \"text\" : \"23 hours 21 mins\", \"value\" : 84081 }, \"status\" : \"OK\" } ] }, { \"elements\" : [ { \"distance\" : { \"text\" : \"938 mi\", \"value\" : 1509173 }, \"duration\" : { \"text\" : \"1 day 2 hours\", \"value\" : 94436 }, \"status\" : \"OK\" }, { \"distance\" : { \"text\" : \"1 ft\", \"value\" : 0 }, \"duration\" : { \"text\" : \"1 min\", \"value\" : 0 }, \"status\" : \"OK\" }, { \"distance\" : { \"text\" : \"1,276 mi\", \"value\" : 2053949 }, \"duration\" : { \"text\" : \"1 day 17 hours\", \"value\" : 146217 }, \"status\" : \"OK\" } ] }, { \"elements\" : [ { \"distance\" : { \"text\" : \"876 mi\", \"value\" : 1409170 }, \"duration\" : { \"text\" : \"23 hours 25 mins\", \"value\" : 84308 }, \"status\" : \"OK\" }, { \"distance\" : { \"text\" : \"1,273 mi\", \"value\" : 2048291 }, \"duration\" : { \"text\" : \"1 day 16 hours\", \"value\" : 145790 }, \"status\" : \"OK\" }, { \"distance\" : { \"text\" : \"1 ft\", \"value\" : 0 }, \"duration\" : { \"text\" : \"1 min\", \"value\" : 0 }, \"status\" : \"OK\" } ] } ], \"status\" : \"OK\"}";
```

Comment:
Why do you need apiResponse here? Can it be empty?

Reply:
An empty response leads to an Exception when it's parsed.